### DISCUSSION TITLE
selector based on time?

### DISCUSSION THREAD

**Discussion by faragos** [ngrx/platform#4349] [Q&A] [answered]
I have a component which renders a coupon. This coupon is only valid to a specific time.
I would like to have a selector called like: selectIsValid()

The problem is the select won't get called again unless I store something in the Store, but I don't want to store the current time. It also doesn't change the output because it's memoized.
Should I handle the logic in the component or a service? Or how can I refresh the select when the selector would be invalid?

any ideas appreciated 😃

**Comment by twittwer** [comment, marked as answer]
I would only save a validUntil in the store and check if it's still valid in the component or set a timer in an effect and trigger an action when the coupon gets invalid.

**Reply by faragos** [reply]
Thank you very much I like the idea with the effect and triggering an action. This could solve it without saving something unnecessary in the store  👍
